docs/envsetup/03_devcontainer.md

```diff
@@ -0,0 +1,106 @@
+# Local installation using VSCode Devcontainers extension
+
+If you have a local Docker installation or are happy to install one, the easiest way to work locally with these materials is to use Visual Studio Code's devcontainer feature. This approach provides all the necessary tools and dependencies without requiring manual installation.
+
+## Requirements
+
+To use the devcontainer setup, you'll need:
+
+- [Visual Studio Code](https://code.visualstudio.com/)
+- A local Docker installation, for example:
+  - [Docker Desktop](https://docs.docker.com/get-docker/) (for Windows/macOS)
+  - [Docker Engine](https://docs.docker.com/engine/install/) (for Linux)
+  - [Colima](https://github.com/abiosoft/colima) (alternative for macOS)
+- [Docker Buildx](https://docs.docker.com/build/concepts/overview/#install-buildx) (included in Docker Desktop, but may need separate installation with other Docker setups)
+- [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) for VS Code
+
+Your Docker installation must be running before you attempt to open the devcontainer.
+
+To verify that Docker buildx is available, run:
+
+```bash
+docker buildx version
+```
+
+If this command fails, you'll need to install the buildx extension before proceeding.
+
+## Setup Instructions
+
+Follow these steps to set up your local environment using VS Code devcontainers:
+
+### Install the "Dev Containers" extension in VS Code
+
+- Open VS Code
+- Go to Extensions (Ctrl+Shift+X or Cmd+Shift+X on macOS)
+- Search for "Dev Containers"
+- Click "Install"
+
+![Installing Dev Containers extension in VS Code](img/install_extension.png)
+
+### Clone the repository:
+
+```bash
+git clone https://github.com/nextflow-io/training.git
+cd training
+```
+
+### Open the repository in VS Code:
+
+- Launch VS Code
+- Select **File -> Open Folder** from the menu
+- Navigate to and select the training repository folder you just cloned
+- Click **Open**
+
+### Reopen in Container
+
+When prompted by VS Code to "Reopen in Container", click on it. Alternatively:
```

Comment:
```suggestion
If prompted by VS Code to "Reopen in Container", click on it. Alternatively:
```

People might have disabled it, like I had, i think `If` is clearer than `When` in such a case

Reply:
especially given you have put in `Alternatively`